### PR TITLE
add Path2D and related methods

### PR DESCRIPTION
**Pull request by buntec**
Hi there!

Before submitting a PR containing any Scala changes, please make sure you...

* run `sbt prePR`
* commit changes to `api-reports`

Thanks for contributing!

### CODE REVIEW

**Review by armanbilge — COMMENTED** [review]
Thanks! Do you mind copying some scaladocs from MDN?

File: dom/src/main/scala/org/scalajs/dom/CanvasRenderingContext2D.scala [hunk 1]
```diff
@@ -124,18 +142,28 @@ class CanvasRenderingContext2D extends js.Object {
   /** Fills the subpaths with the current fill style. */
   def fill(): Unit = js.native
 
+  def fill(path: Path2D): Unit = js.native
+
+  def fill(fillRule: String): Unit = js.native
```
Comment (armanbilge) [inline dom/src/main/scala/org/scalajs/dom/CanvasRenderingContext2D.scala:147]: It looks like `fillRule` can be either `"nonzero"` or `"evenodd"` so we should model like other "enums".
- https://github.com/scala-js/scala-js-dom/tree/77703fd15d27df4dea5219ad7b1e523992bd521a/dom/src/main/scala-2/org/scalajs/dom
- https://github.com/scala-js/scala-js-dom/tree/77703fd15d27df4dea5219ad7b1e523992bd521a/dom/src/main/scala-3/org/scalajs/dom

File: dom/src/main/scala/org/scalajs/dom/Path2D.scala [hunk 2]
```diff
@@ -0,0 +1,57 @@
+package org.scalajs.dom
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.JSGlobal
+
+@js.native
+@JSGlobal
+class Path2D extends js.Object {
```
Comment (armanbilge) [inline dom/src/main/scala/org/scalajs/dom/Path2D.scala:8]:
```
The [[Path2D]] interface of the Canvas 2D API is used to declare a path that can then be used on a [[CanvasRenderingContext2D]] object.
```

File: dom/src/main/scala/org/scalajs/dom/Path2D.scala [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      anticlockwise: Boolean): Unit = js.native
+
+  def arc(x: Double, y: Double, radius: Double, startAngle: Double, endAngle: Double): Unit = js.native
+
+  /** Adds an elliptical arc to the path which is centered at (x, y) position with the radii radiusX and radiusY
+    * starting at startAngle and ending at endAngle going in the given direction by counterclockwise (defaulting to
+    * clockwise).
+    */
+  def ellipse(x: Double, y: Double, radiusX: Double, radiusY: Double, rotation: Double, startAngle: Double,
+      endAngle: Double, anticlockwise: Boolean = js.native): Unit = js.native
```
Comment (armanbilge) [inline dom/src/main/scala/org/scalajs/dom/Path2D.scala:52]: Seems to be called `counterclockwise` in the spec.

```suggestion
      endAngle: Double, counterclockwise: Boolean = js.native): Unit = js.native
```

File: dom/src/main/scala/org/scalajs/dom/Path2D.scala [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  /** Adds a circular arc to the path with the given control points and radius, connected to the previous point by a
+    * straight line.
+    */
+  def arcTo(x1: Double, y1: Double, x2: Double, y2: Double, radius: Double): Unit = js.native
+
+  /** Adds an arc to the path which is centered at (x, y) position with radius r starting at startAngle and ending at
+    * endAngle going in the given direction by counterclockwise (defaulting to clockwise).
+    */
+  def arc(x: Double, y: Double, radius: Double, startAngle: Double, endAngle: Double,
+      anticlockwise: Boolean): Unit = js.native
```
Comment (armanbilge) [inline dom/src/main/scala/org/scalajs/dom/Path2D.scala:43]:
```suggestion
  def arc(x: Double, y: Double, radius: Double, startAngle: Double, endAngle: Double,
      counterclockwise: Boolean): Unit = js.native
```

File: dom/src/main/scala/org/scalajs/dom/CanvasRenderingContext2D.scala (hunk 1 above)
Comment (buntec) [reply dom/src/main/scala/org/scalajs/dom/CanvasRenderingContext2D.scala]: Oh, this is neat!

**Review by armanbilge — APPROVED** [review]
Thanks a bunch!!

**Pull request merged**